### PR TITLE
Add DDR support for varHandleMethodTypeLookupTable

### PR DESCRIPTION
**Pull request by babsingh**
Related: https://github.com/eclipse-openj9/openj9/pull/18976#issuecomment-1954789183

### CODE REVIEW

**Comment by babsingh** [comment]
@keithc-ca Requesting your review.

File: debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/tools/ddrinteractive/RomClassWalker.java [hunk 1]
```diff
@@ -773,6 +779,42 @@ void allSlotsInIntermediateClassDataDo () throws CorruptDataException
...
+					for (int i = 0; i < count; i++) {
+						classWalkerCallback.addSlot(clazz, SlotType.J9_U16, cursor.add(i), "cpIndex");
+					}
+				}
+			} catch (CorruptDataException e) {
+				throw e;
+			} catch (Throwable e) {
+				throw new CorruptDataException(e);
+			}
+		}
```
[the remark this answers is not in the document]
Comment (babsingh) [reply debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/tools/ddrinteractive/RomClassWalker.java:815]: > if (J9BuildFlags.J9VM_OPT_METHOD_HANDLE && J9BuildFlags.J9VM_OPT_OPENJDK_METHODHANDLE) {
> Normally the flags have opposite values, but that's another way a system dump might be corrupted.

We never set both the flags; so, the above condition will never be true in normal (non-corrupt) scenarios. Are we supposed to run the code only in the corrupt cases? or is there a typo: `... && !J9BuildFlags.J9VM_OPT_OPENJDK_METHODHANDLE)`?

[the remark this answers is not in the document]
Comment (babsingh) [reply debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/tools/ddrinteractive/RomClassWalker.java]: All feedback has been incorporated.

**Review by keithc-ca — APPROVED** [review]

**Comment by keithc-ca** [comment]
Jenkins compile amac jdk11,jdk21

**Pull request merged**